### DISCUSSION TITLE
Github Actions doesn't allow basic search in logs

### DISCUSSION THREAD

**Discussion by fmigneault** [community/community#3236] [Actions]
I noticed that whenever we are viewing the output logs of a Workflow job, the basic browser search seems to limit itself to the viewport scope. So if a log as thousands of lines, only the ~50-60 visible lines (or whatever according to your screen size) + roughly 20 on top/bottom are actually searchable. 

For example in below image, I enter `ignore` search. 
In the current area that is visible in the logs (between L1648-1677), there are no matches.

![image](https://user-images.githubusercontent.com/19194484/105880639-9787e000-5fd1-11eb-84d0-16a6c0cffcb6.png)

Then, I scroll up such that L1590-1639 are visible (without changing the search terms). 
The closest `ignore` matches to the viewport where the original search was done (L1648-1677) are found, but all the "overflow" lines on top/bottom are not found! 

![image](https://user-images.githubusercontent.com/19194484/105880830-cd2cc900-5fd1-11eb-92b2-fb8494b864bf.png)

Needless to say that not being able to search through the entire content of the massive logs is extremely counter productive. 
I can't go through all the lines each time an error occurs to attempt finding the source. 

I'm using Firefox 84.0.2 (64-bit), not sure if same applies on other browsers.

**Comment by evi-liu** [comment]
This space is currently reserved for feedback on Discussions and GitHub for Mobile. This is noted in our [README](https://github.com/github/feedback#github-public-feedback-discussions) and the [pinned discussion](https://github.com/github/feedback/discussions/1).

All other feedback at this time can be shared through our [support channel](https://support.github.com/contact/feedback) or our [community forum](https://github.community/).

**Reply by rajbos** [reply]
Given that this space now is open for all feedback, would like a revisit ;-)

**Reply by DevJamesC** [reply]
TIL: If you click the gear next to the "search logs" bar in the top right, there is an option to download the log or view raw log. The raw log allows a full log search using standard browser search.

**Reply by rmunn** [reply]
The raw log appears to be pretty useless when your current (still-running) build is a matrix build; here's the entirety of the raw logs from my current workflow run where I was trying to search for the name of one particularly flaky test:

```plaintext
2022-08-25T08:43:20.9997095Z Requested labels: windows-latest
2022-08-25T08:43:20.9997146Z Job defined at: (ORG)/(REPO)/.github/workflows/dotnet.yml@refs/heads/master
2022-08-25T08:43:20.9997175Z Waiting for a runner to pick up this job...
2022-08-25T08:43:22.0000856Z Job is waiting for a hosted runner to come online.
2022-08-25T08:44:06.0775252Z Job is about to start running on the hosted runner: Hosted Agent (hosted)
```

So I can't tell whether that one test failed until the entire run (which takes 30 minutes as we have some slow tests in there) is finished. Pretty frustrating experience not having *any* way to search the logs until they're complete.
